spec/models/rubric_criterion_spec.rb

```diff
@@ -2,4 +2,233 @@
+      expect(@criterion.criterion_ta_associations.count).to eq(1)
+      @criterion.reload
+      expect(ta2.id).to be == @criterion.tas[0].id
+    end
+
+    it 'get the names of TAs assigned to it' do
+      ta1 = create(:ta)
+      ta2 = create(:ta)
+      @criterion.add_tas(ta1)
+      @criterion.add_tas(ta2)
+      expect(@criterion.get_ta_names).to contain_exactly(ta1.user_name, ta2.user_name)
+    end
+
+  end
+
+  context 'from an assignment without criteria' do
+    before(:each) do
+      @assignment = create(:assignment)
+    end
+
+    context 'when parsing a CSV file' do
+      describe 'raise csv line error on an empty row' do
+        it 'raises' do
+          expect {
+            RubricCriterion.create_or_update_from_csv_row([], @assignment).to raise_error(CSVInvalidLineError)
+          }
+        end
+      end
+    end
+
+    context 'when parsing a CSV file' do
+      describe 'raise csv line error on a 1 element row' do
+        it 'raises' do
+          expect {
+            rubric.create_or_update_from_csv_row(%w(name), @assignment).to raise_error(CSVInvalidLineError)
+          }
+        end
+      end
+    end
+
+    context 'when parsing a CSV file' do
+      describe 'raise csv line error on a 2 element row' do
+        it 'raises' do
+          expect {
+            rubric.create_or_update_from_csv_row(%w(name 1.0), @assignment).to raise_error(CSVInvalidLineError)
+          }
+        end
+      end
+    end
+
+    context 'when parsing a CSV file' do
+      describe 'raise csv line error on rows with elements without names for every criterion' do
+        row = %w(name 1.0)
+        (0..RubricCriterion::RUBRIC_LEVELS - 2).each do |i|
+          row << 'name' + i.to_s
+          it 'raises' do
+            expect {
+              RubricCriterion.create_or_update_from_csv_row(row, @assignment).to raise_error(CSVInvalidLineError)
+            }
+          end
+        end
+      end
+    end
+
+    context 'when parsing a CSV file' do
+      describe 'raise csv line error on a row with an invalid weight' do
+        row = %w(name max_mark l0 l1 l2 l3 l4)
+        it 'raises' do
+          expect {
+            RubricCriterion.create_or_update_from_csv_row(row, @assignment).to raise_error(CSVInvalidLineError)
+          }
+        end
+      end
+    end
+
+    context 'and the row is valid' do
+      before(:each) do
+        # we'll need a valid assignment for those cases.
+        @assignment = create(:assignment)
+        @row = ['criterion 5', '1.0']
+        (0..RubricCriterion::RUBRIC_LEVELS - 1).each do |i|
+          @row << 'name' + i.to_s
+        end
+        # ...containing commas and quotes in the descriptions
+        (0..RubricCriterion::RUBRIC_LEVELS - 1).each do |i|
+          @row << 'description' + i.to_s + ' with comma (,) and ""quotes""'
+        end
+        @csv_base_row = @row
+      end
+    end
+
+    context 'be able to create a new instance without level descriptions' do
```

Comment:
Your problem here is that the `before(:each)` above isn't triggering because it's in a separate `context` block. You need to make this test a `should` rather than a `context`, and make sure it's nested within the `context` block above. Same thing with the next test below (starting on line 221).